cognite/pygen/utils/external_id_factories.py

```diff
@@ -37,116 +66,193 @@ def windmill_factory(domain_cls: type, data: dict) -> str:
 
 """
 
-import uuid
+import logging
+import uuid as uuid_
 from collections import defaultdict
-from hashlib import sha256
-from typing import Any, Callable
+from hashlib import sha256 as sha256_
+from typing import Callable, Optional
+
+logger = logging.getLogger(__name__)
 
 
-def create_uuid_factory(shorten: bool = True) -> Callable[[type, dict], str]:
+def shorten_string(input_str: str, length: int = 7) -> str:
     """
-    This creates a uuid external id factory.
+    This shortens the input string to the specified length.
 
     Args:
-        shorten: If True, the external id will be shortened to 7 characters.
-
+        input_str: The string to shorten
+        length: The length to shorten the input string to
 
     Returns:
-        A factory function that can be set on the DomainModelApply class.
+        The shortened string
     """
+    return input_str[:length]
 
-    def uuid_factory_wrapped(domain_cls: type, data: dict) -> str:
-        """
-        This creates an uuid external id for each domain class.
-        Args:
-            domain_cls:
-            data:
 
-        Returns:
+def domain_name(domain_cls: type, data: dict) -> str:
+    """
+    This creates an string based on the domain class name.
 
-        """
-        prefix = domain_cls.__name__.casefold().removesuffix("apply")
-        uuid_ = str(uuid.uuid4())
-        if shorten:
-            return f"{prefix}:{uuid_[:7]}"
-        else:
-            return f"{prefix}:{uuid_}"
+    Args:
+        domain_cls: The domain class
+        data: The data for the domain class instance
 
-    return uuid_factory_wrapped
+    Returns:
+        A string based on the domain class name
+    """
+    return domain_cls.__name__.casefold().removesuffix("write")
 
 
-def uuid_factory(domain_cls: type, data: dict) -> str:
+def uuid(domain_cls: type, data: dict) -> str:
     """
-    This creates an uuid external id for each domain class.
+    This creates a uuid.
+
     Args:
-        domain_cls:
-        data:
+        domain_cls: The domain class
+        data: The data for the domain class instance
 
     Returns:
+        A uuid
+    """
+    return uuid_.uuid4().hex
 
+
+def sha256(domain_cls: type, data: dict) -> str:
     """
-    return f"{domain_cls.__name__.casefold().removesuffix('apply')}:{uuid.uuid4()}"
+    This creates a sha256 hash based on the data.
 
+    Args:
+        domain_cls: The domain class
+        data: The data for the domain class instance
 
-def create_sha256_factory(shorten: bool = True) -> Callable[[type, dict], str]:
+    Returns:
+        A sha256 hash
     """
-    This creates a sha256 hash external id factory.
+    return sha256_(str(data).encode()).hexdigest()
+
+
+INCREMENTAL_ID_REGISTRY: dict[str, int] = defaultdict(int)
```

Comment:
Shouldn't this be private? In addition, what happens if you instantiate two different `incremental_id` factories? Will they not have a shared state then?
```suggestion
_INCREMENTAL_ID_REGISTRY: dict[str, int] = defaultdict(int)
```

Reply:
Yes, it would share it depending on how it's been imported, but usage of this for external_ids outside anything but testing should be highly discouraged since there's too much potential for conflict in external_ids.